### ISSUE TITLE
Request Timeout Config

### ISSUE THREAD

**Issue by jaliase** [promregator/promregator#181]
## Environment
Promregator version: 0.6.4
CF API Version: 2.124.0

## Summary / Problem Statement
Promregator doesn't honor the configured request timeout in the scheduled task. The timeout is hardcoded to 2500ms.

## Observed Behavior
14:55:35.651: [APP/PROC/WEB.0] 2020-05-29 18:55:35.650  WARN 9 --- [     parallel-4] o.c.p.cfaccessor.ReactiveCFAccessorImpl  : Woof woof! It appears that the connection to the Cloud Controller is gone. Trying to restart Cloud Foundry Client
14:55:35.651: [APP/PROC/WEB.0] java.util.concurrent.TimeoutException: Did not observe any item or terminal signal within 2500ms in 'Mono.checkpoint â¢ org.cloudfoundry.reactor.client.v2.info.ReactorInfo.get(ReactorInfo.java:47)' (and no fallback has been configured)

## Expected Behavior
Promregator should use the configured timeout.

**Comment by eaglerainbow** [comment]
Hi jaliase,

Welcome to Promregator. And thanks for sharing your experience with us!

Yes, you are right: The request timeout for fetching a forced restart of the connection pool to the CF Cloud Connector (CFCC) cannot be configured right now. I did so, because none of the existing request timeouts configured would fit for this type of request. We could introduce another one, but already describing it in a documentation, what that parameter does, makes you feel strange. 

There are two things which also make me a little wonder:

1. Having a timeout on the `/info` endpoint of your platform taking more than 2.5 seconds is really strange: Such a bad roundtrip time from your platform would indicate that either the platform itself or the communication between Promregator and the platform is really broken. Usually, platforms should reply on this endpoint within milliseconds.

2.  You state that you are still using Promregator 0.6.4. Note that @checketts found a major and very complicated bug in the underlying communication library to CFCC. This was fixed with [0.7.3](https://github.com/promregator/promregator/releases/tag/v0.7.3), but wasn't downported to 0.6.x (current version on the 0.7 track is 0.7.5). It appears reasonable to me that the symptom you describe above *may* be fixed by this correction as well (the symptoms of that issue can have numerous faces - one of them are strange request timeouts). 


So, long story short: I see little point in adding an additional configuration parameter for the request timeout of the `/info` endpoint as of now. Instead, I would prefer to first go into the direction finding out why the (high) timeout of 2500ms is reached instead. Perhaps you are experiencing an already known bug, which we already could have fixed. 
That is why I would kindly ask you to re-check your issue with a version higher than 0.7.3 (recommended: 0.7.5). If you are still experiencing the info-timeout issue there as well, we can still discuss introducing a request timeout afterwards.

HTH.

**Comment by eaglerainbow** [comment]
BTW: I forgot to mention that with ad3773a20f012433f5d7254c5f4e6a4af8b84dea (currently in the 0.8.x track only), there are some capabilities included, which will permit you to also configure the watchdog's behavior.

**Comment by jaliase** [comment]
Thanks for the quick response. 

So our CF environment is currently overloaded. We have tested 0.6.4 in other CF env with decent load and it works fine. Timeouts are OK we were more concerned with the java.lang.OutOfMemoryError: Direct buffer memory.

Let me check the defect and test with version 0.7.6. I will update you.

**Comment by eaglerainbow** [comment]
> Timeouts are OK we were more concerned with the java.lang.OutOfMemoryError: Direct buffer memory.

Yeah. It's a typical symptom for the issue checketts had found and fixed. Though not having done detailed analysis of your exact case, still all what you said fits very well into the picture of the internals of the issue. That is why I would be rather confident that version 0.7.6 should do the trick. If, against all odds, this should not be the case, please feel free to report details here such that we can follow up on it.
